sonaranalyzer-dotnet/rspec/cs/S5547_c#.html

```diff
@@ -1,18 +1,32 @@
 <p><a href="https://en.wikipedia.org/wiki/Strong_cryptography">Strong cipher algorithms</a> are cryptographic systems resistant to cryptanalysis, they
 are not vulnerable to well-known attacks like brute force attacks for example. </p>
-<p>It is recommended to use only cipher algorithms intensively tested and promoted by the cryptographic community.</p>
+<p>A general recomandation is to only use cipher algorithms intensively tested and promoted by the cryptographic community.</p>
+<p>More specifically for block cipher, it's not recommended to use algorithm with a block size inferior than 128 bits.</p>
 <h2>Noncompliant Code Example</h2>
+<p>For <a href="https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography?view=netcore-3.1">System.Security.Cryptography</a> library,
+these old cryptographic algorithms should no longer be used for any reason:</p>
 <pre>
 var tripleDES1 = new TripleDESCryptoServiceProvider(); // Noncompliant: Triple DES is vulnerable to meet-in-the-middle attack
 
 var simpleDES = new DESCryptoServiceProvider(); // Noncompliant: DES works with 56-bit keys allow attacks via exhaustive search
 
 var RC2 = new RC2CryptoServiceProvider(); // Noncompliant: RC2 is vulnerable to a related-key attack
 </pre>
+<p>For Bouncycastle library, <a href="https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-1000339">AESFastEngine has a side channel leak</a>, it
```

Comment:
I'm not sure if this is tied to new functionality, I'll ask @eric-therond-sonarsource

Reply:
From what I can see in the code only https://rules.sonarsource.com/csharp/RSPEC-4426 has checks for Bouncycastle types.

Reply:
these changes are related to this ticket: https://github.com/SonarSource/sonar-dotnet/issues/3330